File: MePlusPlus/fancy/src/main/java/com/meplus/fancy/model/Response.java
package com.meplus.fancy.model;

/**
 * Created by dandanba on 3/9/16.
 */
public class Response<T> {
    public String getCode() {
        return Code;
    }

    public void setCode(String code) {
        Code = code;
    }

    public String getMessage() {
        return Message;
    }

    public void setMessage(String message) {
        Message = message;
    }

    public T getResult() {
        return Result;
    }

    public void setResult(T result) {
        Result = result;
    }

    private String Code;
    private String Message;
    private T Result;
}

File: MePlusPlus/fancy/src/main/kotlin/com/meplus/fancy/model/Response.kt
package com.meplus.fancy.model

import android.content.Context
import cn.trinea.android.common.util.ToastUtils
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Created by dandanba on 4/11/16.
 */
data class Response<T>(
        @JsonProperty("Code")
        var Code: String ? = null,
        @JsonProperty("Message")
        var Message: String ? = null,
        @JsonProperty("Result")
        var Result: T ? = null
) {
    fun isOk(): Boolean {
        return "ok".equals(Code!!.toLowerCase());
    }

    fun showMessage(context: Context) {
        ToastUtils.show(context, Message);
    }
}